src/internal.c

```diff
@@ -228,15 +228,25 @@ static WC_INLINE int IsEncryptionOn(WOLFSSL* ssl, int isSend)
  * If SCTP is enabled returns dtls && !sctp. */
 static WC_INLINE int IsDtlsNotSctpMode(WOLFSSL* ssl)
 {
-    int result = ssl->options.dtls;
-
-    if (result) {
 #ifdef WOLFSSL_SCTP
-        result = !ssl->options.dtlsSctp;
+    return ssl->options.dtls && !ssl->options.dtlsSctp;
+#else
+    return ssl->options.dtls;
 #endif
-    }
+}
 
-    return result;
+int IsInitialRenegotiationState(WOLFSSL* ssl)
+{
+    if (ssl->options.acceptState == ACCEPT_FIRST_REPLY_DONE
```

Comment:
How does ACCEPT_FIRST_REPLY_DONE guarantee Renegotiation?

Reply:
`IsInitialRenegotiationState` is used in `DoClientHello` to determine whether or not the `HelloVerifyReqeust` is necessary. I decided (with John's approval) that a `HelloVerifyReqeust` is unnecessary for a renegotiation. In `DoHandShakeMsgType` if we receive a `client_hello` and have `ssl->secure_renegotiation = 1` then `ssl->options.acceptState` is set to `ACCEPT_FIRST_REPLY_DONE`. From testing this is a sufficient signal for `DoClientHello` to know that it doesn't need to compute the Hmac.

Reply:
But this function doesn't test secure_renegotation status.  We shouldn't rely on someone testing that first either.